src/analysis/buildTypeWeights.ts

```diff
@@ -112,9 +113,33 @@ function parseObjectFields(
                     resolveTo: listType.toString().toLocaleLowerCase(),
                 };
             } else {
+                // if the @listCost directive is given for the field,
+                // apply the cost argument's value to the field's weight
+                const directives = fields[field].astNode?.directives;
+
+                if (directives && directives.length > 0) {
+                    // eslint-disable-next-line consistent-return
+                    directives.forEach((dir) => {
+                        if (dir.name.value === 'listCost') {
+                            if (dir.arguments && dir.arguments[0])
+                                result.fields[field] = {
```

Comment:
if you change line 124 to be `if (dir.arguments && dir.arguments[0].value.kind === Kind.INT)` you don't need the `ts-ignore`, `ts-error` or `eslint` escapes on lines 128-130 and it is more strongly typed require an `Int` which is possible a bug.

Reply:
THANK YOU